Gems/MotionMatching/Code/Source/FeaturePosition.cpp

```diff
@@ -54,12 +54,17 @@ namespace EMotionFX::MotionMatching
     void FeaturePosition::DebugDraw(AzFramework::DebugDisplayRequests& debugDisplay,
         const Pose& currentPose,
         const FeatureMatrix& featureMatrix,
+        const FeatureMatrixTransformer* featureTransformer,
         size_t frameIndex)
     {
         const Transform jointModelTM = currentPose.GetModelSpaceTransform(m_jointIndex);
         const Transform relativeToWorldTM = currentPose.GetWorldSpaceTransform(m_relativeToNodeIndex);
 
-        const AZ::Vector3 position = GetFeatureData(featureMatrix, frameIndex);
+        AZ::Vector3 position = GetFeatureData(featureMatrix, frameIndex);
+        if (featureTransformer)
+        {
+            position = featureTransformer->InverseTransform(position, m_featureColumnOffset);
+        }
```

Comment:
question: Would it be worth wrapping this logic so it doesn't just happen in the DebugDraw code itself (here and in `FeatureVelocity`). It just feels having a routing to do this in the core logic (if anything else needs to get the accurate position) would be nicer than having to remember to do this check and update `position` or `velocity`). Not a huge deal just feels like a potential maintenance trip-up

Reply:
I think it is good to be explicit here, elsewise we'd have to embed some hint into the function name, making that long. I'd find that more confusing somehow.